src/Swarm/Language/Text/Markdown.hs

```diff
@@ -181,7 +216,8 @@ fromTextPure :: Text -> Either String (Document Text)
 fromTextPure t = do
   let spec = Mark.rawAttributeSpec <> Mark.defaultSyntaxSpec <> Mark.rawAttributeSpec
   let runSimple = left show . runIdentity
-  runSimple $ Mark.commonmarkWith spec "markdown" t
+  Document tokenizedDoc <- runSimple $ Mark.commonmarkWith spec "markdown" t
+  return . Document $ normalise <$> tokenizedDoc
```

Comment:
Random thought: would there be any value in having a `NormalizedDocument` newtype whose "smart constructor" performs the `normalize` function?

Reply:
Honestly, if `Document` used `Seq` I would consider doing it immediately in `(<>)`, the naive approach here merges the text one-by-one anyway.